app/PickMe_nebula0/app/src/main/java/com/example/pickme_nebula0/event/EventDetailUserActivity.java

```diff
@@ -58,53 +64,66 @@ protected void onCreate(Bundle savedInstanceState) {
             finish();
         }
 
-        // Link components
+        // link main components
         backButton = findViewById(R.id.button_edu_back);
-        acceptButton = findViewById(R.id.button_edu_accept);
-        declineButton = findViewById(R.id.button_edu_decline);
         unregisterButton = findViewById(R.id.button_edu_unregister);
         eventDetailsTextView = findViewById(R.id.textView_edu_details);
         userStatusTextView = findViewById(R.id.textView_edu_status);
         registerButton = findViewById(R.id.button_edu_reg);
 
-        // Initially set all buttons invisible (may take a second to query DB and update visibility)
-        acceptButton.setVisibility(View.GONE);
-        declineButton.setVisibility(View.GONE);
+        // link notification components
+        notificationLayout = findViewById(R.id.notification_layout);
+        acceptButton = findViewById(R.id.entrant_event_notification_accept);
+        declineButton = findViewById(R.id.entrant_event_notification_decline);
+        notificationMessage = findViewById(R.id.entrant_event_notification_message);
+
+        // Initially set main buttons invisible (may take a second to query DB and update visibility)
         unregisterButton.setVisibility(View.GONE);
         registerButton.setVisibility(View.GONE);
 
+        // initially set notification invisible, but components to visible
+        notificationLayout.setVisibility(View.GONE);
+        acceptButton.setVisibility(View.VISIBLE);
+        declineButton.setVisibility(View.VISIBLE);
+        notificationMessage.setVisibility(View.VISIBLE);
+
+        // when back button clicked, finish activity
         backButton.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View v) {
                 finish();
             }
         });
 
+        // when register button clicked, user joins waitlist
         registerButton.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View v) {
                 joinWaitlist(eventID, () -> { fromQR = false; renderAll(); });
             }
         });
 
-        acceptButton.setOnClickListener(new View.OnClickListener() {
+        // when unregister button clicked, user status set to canceled
+        unregisterButton.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View v) {
-                dbManager.setRegistrantStatus( eventID, userID, DBManager.RegistrantStatus.CONFIRMED);
-                renderUserStatus(); // status has changed so re-render
+                // TODO - we can change this later to fully remove them if we don't want a user of leaves of their own volition CANCELLED
```

Comment:
let's leave this as CANCELLED since the organizer needs to be able to send notifications to cancelled entrants